x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/index.tsx

```diff
@@ -26,20 +28,34 @@ const SystemPromptComponent: React.FC<Props> = ({
   onSystemPromptSelectionChange,
   setIsSettingsModalVisible,
   allSystemPrompts,
+  refetchConversations,
 }) => {
+  const [isCleared, setIsCleared] = useState(false);
```

Comment:
Just commenting that UX is a little different from the default `EuiSuperSelect` behavior, as when you click to select an item the checkbox is still there as if it were already selected. This component is way more complex than it needs to be as you mentioned in the description, so this fix you provided is more than sufficient from my perspective. 

That said, we should probably clean this up after the release -- we need to swap over to not sending the prompt as part of the user message (and fetch it server side now that they're persisted), so we could do this as part of that work. Thanks for working around the leftover tangled 🍝 here in this component. 🙏 


<p align="center">
  <img width="300" src="https://github.com/user-attachments/assets/77d8040c-e8af-42c7-8e25-2ee671b3230a" />
</p>

Reply:
I agree, the code here can be simplified quite a bit by combining `system_prompt` and `select_system_prompt`. But a bigger refactor would include `impl/assistant/index.tsx` which is in bad shape right now with all the state changes, I'm sure there are uncaught side effects 🙈